src/components/HomePage/index.tsx

```diff
@@ -1,21 +1,19 @@
+import { get } from "lodash";
 import * as React from "react";
 import { Query } from "react-apollo";
 import { Link } from "react-router-dom";
 
-import { Button, Carousel, Loader, ProductListItem } from "..";
+import { Button, Carousel, Loader, ProductListItem, ProductsFeatured } from "..";
 import { ProductsList } from "../../core/types/saleor";
-import { generateCategoryUrl, generateProductUrl } from "../../core/utils";
+import { generateCategoryUrl } from "../../core/utils";
 import { Error } from "../Error";
 import { GET_PRODUCTS_AND_CATEGORIES } from "./queries";
 
 import "./scss/index.scss";
 
 const canDisplay = (data: ProductsList) =>
-  data &&
-  data.shop &&
-  data.shop.homepageCollection &&
-  data.categories &&
-  data.categories.edges;
+  get(data, "shop.homepageCollection") &&
```

Comment:
It would be better if we used "maybe" monad:
```
function maybe<T>(exp: () => T, d?: T) {
  try {
    const result = exp();
    return result === undefined ? d : result;
  } catch {
    return d;
  }
}
```

Passing string means that TS would not check if `data`'s type has those keys, so in case keys have changed, we would have to remember to change this instead of getting big-ass error during type checking.